rate: add prometheus definitions, docs, and clearer names (#15945)
hashicorp/consul: website/content/docs/agent/telemetry.mdx

| `consul.rpc.accept_conn`                            | Increments when a server accepts an RPC connection.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | connections                       | counter |

> | `consul.rpc.rate_limit.exceeded`                    | Increments whenever an RPC is over a configured rate limit. Note: in permissive mode, the RPC will have still been allowed to proceed.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | RPCs                              | counter |

Reviewer: ```suggestion
| `consul.rpc.rate_limit.exceeded`                    | Increments whenever an RPC is over a configured rate limit. When Consul is in `permissive` mode, the RPC is allowed to proceed.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | RPCs                              | counter |
```
Actually, what is in permissive mode in this context? I assumed that `permissive` is a value that you you can define in some configuration, but I'm not sure. Also, we should just use simple present tense per the style guide.

Author: Thanks for fixing my terrible writing! 😆 ❤️  

In this context, the rate limiter is in permissive mode, and there can be multiple limiters involved in a single RPC (e.g. global, ip-based, tenant-based) which are configured independently.
